### DISCUSSION TITLE
Which platform or framework do you use for presenting your visualizations?

### DISCUSSION THREAD

**Discussion by timbmg** [vega/altair#3293] [Q&A]
Hi 👋 

I am wondering which platform or framework you use to present your visualizations?

Of course I can just export as png and paste it to some google slides or powerpoint. But I am looking for something nicer, ideally that keep the plots interactive.

**Comment by ChristopherDavisUCI** [comment]
I think I've used these four approaches before to share interactive Altair charts.  I'm curious to hear what others usually do:

1.  Share a link to the chart in the Vega Editor.
2.  Put the chart in an html tile as described [here](https://altair-viz.github.io/user_guide/saving_charts.html#html-format) and post that on the web.
3. In a Jupyter-Book project (which is in the same spirit as how the docs are made).
4. As part of a Streamlit app, using `st.altair_chart`.

I'm happy to give more details on any of those if it would be helpful.

**Comment by joelostblom** [comment]
I usually present any non interactive charts as images in Google slides, and then have links/other tabs open with local HTML files if there is a more complex chart with interactivity that I need to walk through.

I prefer non-programmatic slide editors in general when making presentations and I don't know any such editor that support interactive charts. However, it it possible to use programmatic slide editors such as Jupyter RISE, RevealJS, and Quarto to have interactive charts show up together with the rest of the content in the slides without switching to another tab.

**Comment by binste** [comment]
I usually share Altair visualizations either:

* static as svg/png
* interactive and standalone (i.e. no webserver) with [Quarto](https://quarto.org/docs/get-started/hello/jupyter.html) where I'd write for example a report/analysis in Jupyter notebooks and then convert it to a single HTML file which is easy to share. Can highly recommend it. This will preserve interactivity of Altair charts and you get a great design out of the box which you can further modify as you see fit (include a company logo, etc.). [Here](https://gist.github.com/binste/201f6e32b1db509d613265abfdc0d001) are the Quarto settings I usually use.
* interactive in a web application. I work with [Plotly Dash](https://dash.plotly.com/) (using [dash-vega-components](https://github.com/altair-viz/dash-vega-components)), [Streamlit](https://docs.streamlit.io/library/api-reference/charts/st.altair_chart) (as previously mentioned). [Panel](https://panel.holoviz.org/reference/panes/Vega.html) also looks nice.

**Comment by afonit** [comment]
I run a [Voila](https://github.com/voila-dashboards/voila) on an Ubuntu VM on the internal infrastructure of the company I work for. I have a git repo which I push all my notebooks to which get served on the VM.  Usually there are a lot of ipywidgets and database queries - hence the need to run this through Voila on a vm.

**Comment by AlloriMD** [comment]
I use Altair in Jupyter notebooks for data exploration and Quarto[^1] for published reports. I am also testing Altair in Shiny for Python.

[^1]: (Although, presently Altair charts only display in Quarto web sites and are not rendered in PDF output. It would be nice if there were an Altair extension for Quarto that would help with that, maybe by substituting the SVG for the dynamic plots.)
